File: 01-04-2026/nth_fibonacci_number.py
def nth_fibonacci(n):
    fibArr = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55]

    if n <= 11:
        return fibArr[n - 1]
    else:
        for i in range(11, n):
            fibArr.append(fibArr[i - 1] + fibArr[i - 2])

    return fibArr[n - 1]

print(nth_fibonacci(4)) # 2.
print(nth_fibonacci(10)) # 34.
print(nth_fibonacci(15)) # 377.
print(nth_fibonacci(40)) # 63245986.
print(nth_fibonacci(75)) # 1304969544928657.

File: 01-04-2026/nth_fibonacci_number.js
function nthFibonacci(n) {
  const fibArr = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55];

  if (n <= 11) return fibArr[n - 1];
  else for (let i = 11; i < n; i++) fibArr.push(fibArr[i - 1] + fibArr[i - 2]);

  return fibArr[n - 1];
}

console.log(nthFibonacci(4)); // 2.
console.log(nthFibonacci(10)); // 34.
console.log(nthFibonacci(15)); // 377.
console.log(nthFibonacci(40)); // 63245986.
console.log(nthFibonacci(75)); // 1304969544928657.
